Fields/DDUpload.php

```diff
@@ -71,7 +87,7 @@ public static function process(){
 		\JFactory::getApplication()->close(403);
 	}
 
-	public static function onBeforeStore(&$model, $name, $defenition){
+	public static function onBeforeStore(&$model, $name = '', $defenition = array()){
```

Comment:
Why default value?
Shouldn't be called without specifying filed name and definition.
By typo should be corrected `defenition` to `definition`